nucliadb/nucliadb/search/requesters/utils.py

```diff
@@ -241,3 +249,18 @@ def validate_node_query_results(results: list[Any]) -> Optional[HTTPException]:
             return HTTPException(status_code=status_code, detail=reason)
 
     return None
+
+
+def debug_nodes_info(
+    nodes: list[tuple[AbstractIndexNode, str]]
+) -> list[dict[str, str]]:
+    details: list[dict[str, str]] = []
+    for node, shard_id in nodes:
+        info = {
+            "id": node.id,
+            "shard_id": shard_id,
+            "address": node.address,
+        }
+        if node.primary_id:
+            info["primary_id"] = node.primary_id
```

Comment:
forgot append the `info` to the `details`. I would add a test  for this too, as it may be pretty important info to have for debugging

Reply:
what about adding an integration test that shows what is returned on the http response?

Reply:
Oh dang, thanks. I'll add a test